src/main/java/org/radarcns/management/domain/MetaToken.java

```diff
@@ -0,0 +1,158 @@
+package org.radarcns.management.domain;
+
+import java.time.Instant;
+import java.util.Objects;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.EntityListeners;
+import javax.persistence.FetchType;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.SequenceGenerator;
+import javax.persistence.Table;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Pattern;
+
+import org.apache.commons.lang3.RandomStringUtils;
+import org.hibernate.annotations.Cache;
+import org.hibernate.annotations.CacheConcurrencyStrategy;
+import org.hibernate.envers.Audited;
+import org.radarcns.auth.config.Constants;
+import org.radarcns.management.domain.support.AbstractEntityListener;
+
+@Entity
+@Audited
+@Table(name = "radar_meta_token")
+@Cache(usage = CacheConcurrencyStrategy.NONSTRICT_READ_WRITE)
+@EntityListeners({AbstractEntityListener.class})
+public class MetaToken extends AbstractEntity {
+
+
+    //https://math.stackexchange.com/questions/889538/
+    // probability-of-collision-with-randomly-generated-id
+    // Current length of tokenName is 12. If we think there might be collision we can increase
+    // the length.
+    private static final int SHORT_ID_LENGTH = 12;
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "sequenceGenerator")
+    @SequenceGenerator(name = "sequenceGenerator", initialValue = 1000)
+    private Long id;
+
+    @NotNull
+    @Pattern(regexp = Constants.TOKEN_NAME_REGEX)
+    @Column(name = "token_name", nullable = false, unique = true)
+    private String tokenName;
+
+    @Column(name = "token", length = 2000)
+    private String token;
+
+    @Column(name = "fetched", nullable = false)
+    private Boolean fetched;
+
+    @Column(name = "expiry_date")
+    private Instant expiryDate = null;
+
+    @ManyToOne(fetch = FetchType.EAGER)
+    @JoinColumn(name = "subject_id")
+    @Cache(usage = CacheConcurrencyStrategy.NONSTRICT_READ_WRITE)
+    private Subject subject;
+
+    /**
+     * Meta token constructor.
+     * Generates a random string as the tokenName.
```

Comment:
Perhaps we should rephrase this as something like 'Must generate a random string as the tokenName'. Since other parts in the code rely on the fact that this is happening.

Reply:
We can even add a `MetaTokenUnitTest.java` to check that a random `tokenName` is indeed generated. I'm just a bit concerned that if at some point someone removes the generation of `tokenName` for some reason, it will break `MetaTokenService.saveUniqueToken()`, which will not be immediately apparent otherwise.